app/assets/javascripts/arclight/oembed_viewer.js

```diff
@@ -5,27 +5,45 @@ class OembedLoader {
 
   load() {
     const loadedAttr = this.el.getAttribute('loaded')
-    var data = this.el.dataset
-    var resourceUrl = data.arclightOembedUrl
+    const { arclightOembed, arclightOembedUrl, ...extraOembedParams } = this.el.dataset
+    const extraParams = OembedLoader.normalizeParams(extraOembedParams)
+
     if (loadedAttr && loadedAttr === 'loaded') {
       return
     }
 
-    fetch(resourceUrl)
+    fetch(arclightOembedUrl)
       .then((response) => response.text())
       .then((body) => {
-        const oEmbedEndPoint = OembedLoader.findOEmbedEndPoint(body)
+        const oEmbedEndPoint = OembedLoader.findOEmbedEndPoint(body, extraParams)
         if (!oEmbedEndPoint || oEmbedEndPoint.length === 0) {
           return
         }
         this.loadEndPoint(oEmbedEndPoint)
       })
   }
 
-  static findOEmbedEndPoint(body) {
+  // Convert data-arclight-oembed-* attributes to params for the viewer
+  static normalizeParams(params) {
+    return Object.keys(params).reduce((acc, key) => {
+      // Reverse data attribute name conversion. See:
+      // https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset#name_conversion
+      const newKey = key.replace('arclightOembed', '')
+        .replace(/([a-z])([A-Z])/g, match => `${match[0]}-${match[1].toLowerCase()}`)
+        .toLowerCase()
```

Comment:
Do we need two calls to `toLowerCase()`?

Reply:
The second one handles the case where there is only one word, e.g. `search`. It'll still be capitalized since the original attribute name was `arclightOembedSearch`, so we need to lowercase it after removing `arclightOembed`.

Reply:
Do we need the first one then?

Reply:
I guess not! makes the replacement simpler too.

Reply:
Fixed.